src/main/scala/tech/phosphorus/intellij/prolog/formatting-defs.scala

```diff
@@ -0,0 +1,124 @@
+package tech.phosphorus.intellij.prolog.formatting
+
+import com.intellij.formatting._
+import com.intellij.lang.{ASTNode, Language}
+import com.intellij.psi.TokenType
+import com.intellij.psi.codeStyle.LanguageCodeStyleSettingsProvider.SettingsType
+import com.intellij.psi.codeStyle.{CodeStyleSettings, CodeStyleSettingsCustomizable, LanguageCodeStyleSettingsProvider}
+import com.intellij.psi.formatter.common.AbstractBlock
+import tech.phosphorus.intellij.prolog.PrologLanguage
+import tech.phosphorus.intellij.prolog.formatting.PrologFormattingModelBuilder.createSpaceBuilder
+import tech.phosphorus.intellij.prolog.psi.PrologTypes
+
+import java.util
+import scala.annotation.tailrec
+
+class PrologFormattingModelBuilder extends FormattingModelBuilder {
+  override def createModel(formattingContext: FormattingContext): FormattingModel = {
+    val codeStyleSettings = formattingContext.getCodeStyleSettings
```

Comment:
Since the `codeStyleSettings` is kind of a pure value, you can consider a `let in` style here to make this function a pure value return - see suggestions below.